docs/tangle.md

```diff
@@ -0,0 +1,276 @@
+# Tangle
+
+## Data types
+| Name   | Description   |
+| ------ | ------------- |
+| uint8  | An unsigned 8 bit integer encoded in Little Endian. |
+| uint16  | An unsigned 16 bit integer encoded in Little Endian. |
+| uint32  | An unsigned 32 bit integer encoded in Little Endian. |
+| uint64  | An unsigned 64 bit integer encoded in Little Endian. |
+| ByteArray[N] | A static size array of size N.   |
+| ByteArray | A dynamically sized array. A uint32 denotes its length.   |
+| string | A dynamically sized array of an UTF-8 encoded string. A uint16 denotes its length.   |
+| time    | Unix time in nanoseconds stored as `int64`, i.e., the number of nanoseconds elapsed since January 1, 1970 UTC. |
+
+<summary>Subschema Notation</summary>
+<table>
+    <tr>
+        <th>Name</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td><code>oneOf</code></td>
+        <td>One of the listed subschemas.</td>
+    </tr>
+    <tr>
+        <td><code>optOneOf</code></td>
+        <td>Optionally one of the listed subschemas.</td>
+    </tr>
+    <tr>
+        <td><code>anyOf</code></td>
+        <td>Any (one or more) of the listed subschemas.</td>
+    </tr>
+    <tr>
+        <td><code>between(x,y)</code></td>
+        <td>Between (but including) x and y of the listed subschemas.</td>
+    </tr>
+</table>
+
+## Parameters
+- `MAX_MESSAGE_SIZE=64 KB` The maximum allowed message size.
+- `MAX_PAYLOAD_SIZE=65157 B` The maximum allowed payload size.
+- `MIN_STRONG_PARENTS=1` The minimum amount of strong parents a message needs to have.
```

Comment:
Maybe say that the concept of strong and weak parents will be defined elsewhere?